x-pack/plugins/uptime/public/pages/settings.tsx

```diff
@@ -47,12 +50,16 @@ export interface SettingsFormProps {
   isDisabled: boolean;
 }
 
-const isValidCertVal = (val: string | number) => {
+export const isValidCertVal = (val: string | number) => {
   if (val === '') {
     return Translations.BLANK_STR;
   }
-  if (val === 0) {
-    return VALUE_MUST_BE_GREATER_THEN_ZEO;
+  if (val <= 0) {
+    return VALUE_MUST_BE_GREATER_THAN_ZERO;
+  }
+  const number = Number(val);
+  if (!number || number % 1) {
+    return VALUE_MUST_BE_AN_INTEGER;
   }
```

Comment:
it already does number cast for  vals in on change events in forms, why we need to do it here?

Reply:
The type of `val` is `string | number`. I could add code that only creates a number object if `typeof val` is `string`:

`const number = typeof val === 'string' ? Number(val) : val`.

Reply:
I spoke with @shahzad31 extensively about this offline and the result of our conversation is https://github.com/elastic/kibana/pull/66425/commits/74ac86c66fa9636569d570c65d432934992c9709.

Basically, the code that calls this function does something kind of evil in terms of TS that can result in `val` being `undefined | number`. We [are calling it by spreading the existing settings fields](https://github.com/elastic/kibana/blob/master/x-pack/plugins/uptime/public/pages/settings.tsx#L102), and an object that is `Partial<DynamicSettings>`. 

```typescript
      setFormFields({
        ...formFields,
        ...changedField,
      });
```

This results in our new state can being open to `undefined`. The validator function disables the form's submit input, so we won't allow the user to submit invalid values, but the true answer to this is re-typing the state for `formFields` to be `Partial<DynamicSettings>`. This will expose the areas of code where we're vulnerable to bad values if the validator wasn't catching the cases we are worried about already.

An alternative solution is dropping the value of `change` events on the floor if they don't correspond to our criteria (for cert values this is `!isNaN && !undefined && > 0 && !%1`). I think this would be more confusing for the user because they won't see an error message, or any kind of evidence that their page interactions are registering.